src/prisma/schema.prisma

```diff
@@ -76,20 +76,24 @@ model Tokens {
 }
 
 model WalletDetails {
-  address            String  @id @map("address")
-  type               String  @map("type")
-  label              String? @map("label")
+  address                            String  @id @map("address")
+  type                               String  @map("type")
+  label                              String? @map("label")
   // Local
-  encryptedJson      String? @map("encryptedJson")
+  encryptedJson                      String? @map("encryptedJson")
   // KMS
-  awsKmsKeyId        String? @map("awsKmsKeyId")
-  awsKmsArn          String? @map("awsKmsArn")
+  awsKmsKeyId                        String? @map("awsKmsKeyId") // deprecated and unused, todo: remove with next breaking change. Use awsKmsArn
+  awsKmsArn                          String? @map("awsKmsArn")
+  awsKmsSecretAccessKey              String? @map("awsKmsSecretAccessKey") // if not available, default to: Configuration.awsSecretAccessKey
+  awsKmsAccessKeyId                  String? @map("awsKmsAccessKeyId") // if not available, default to: Configuration.awsAccessKeyId
```

Comment:
I'm not sure if the ROI on letting users set different configuration per wallet is worth the code/DX complexity. While there are use cases this supports, I don't think anyone has saw this as a problem, and the workaround is to make sure your creds have permissions on all regions.

Reply:
It makes sense to have it in this layer of abstraction -- it's a good practice to provision granular scope access key id and secret, so in this case, we can support per kms arn wallet  = 1 aws key. instead of 1 aws key = engine with a wider scope

Reply:
imo, lower-level code abstraction should provide us the flexibility to support wider use case. DX problems can be addresses through a higher level abstraction that glue lower-level interfaces together. This way we can get a good dx + being flexible